ts/nni_manager/training_service/reusable/dlc/dlcClient.ts

```diff
@@ -91,18 +93,40 @@ export class DlcClient {
               ]
         });
         this.log.debug(this.pythonShellClient.command);
-        this.pythonShellClient.on('message', function (envId: any) {
-            // received a message sent from the Python script (a simple "print" statement)
-            deferred.resolve(envId);
-        });
+        this.onMessage();
+        this.log.debug(`on message`);
         this.monitorError(this.pythonShellClient, deferred);
+        this.log.debug(`monitor submit`);
+        const log = this.log;
+        this.pythonShellClient.on('message', (message: any) => {
```

Comment:
Why this is not part of `onMessage()`?

Reply:
because the jobid is call only once in DlcEnvironmentService,
if it is the part of onmessage,DlcEnvironmentService cannot get the jobid